modules/test/playwright/tests/commerce/commerce-theme-minium/main/commerceAccountSelector.spec.ts

```diff
@@ -9,62 +9,255 @@ import {apiHelpersTest} from '../../../../fixtures/apiHelpersTest';
+			});
+
+			layout = await apiHelpers.headlessDelivery.createSitePage({
+				pageDefinition: getPageDefinition([
+					getWidgetDefinition({
+						id: getRandomString(),
+						widgetName:
+							'com_liferay_commerce_order_content_web_internal_portlet_CommerceOpenOrderContentPortlet',
+					}),
+				]),
+				siteId: site.id,
+				title: getRandomString(),
+			});
+
+			channel = await apiHelpers.headlessCommerceAdminChannel.postChannel(
+				{
+					siteGroupId: site.id,
+				}
+			);
+
+			await commerceAdminChannelsPage.changeCommerceChannelSiteType(
+				channel.name,
+				'B2B'
+			);
 		});
 
-		await apiHelpers.headlessCommerceAdminOrder.postOrder({
-			accountId: account.id,
-			channelId: channel.id,
-			name: 'order1',
-			orderStatus: '2',
+		await test.step('Create an Account and a buyer', async () => {
+			account = await apiHelpers.headlessAdminUser.postAccount({
+				name: getRandomString(),
+				type: 'business',
+			});
+
+			const user =
+				await apiHelpers.headlessAdminUser.getUserAccountByEmailAddress(
+					'demo.unprivileged@liferay.com'
+				);
+
+			const rolesResponse =
+				await apiHelpers.headlessAdminUser.getAccountRoles(account.id);
+
+			const accountRoleBuyer = rolesResponse?.items?.filter((role) => {
+				return role.name === 'Buyer';
+			});
+
+			await apiHelpers.headlessAdminUser.assignAccountRoles(
+				account.externalReferenceCode,
+				accountRoleBuyer[0].id,
+				user.emailAddress
+			);
+
+			const siteRole =
+				await apiHelpers.headlessAdminUser.getRoleByName('Site Member');
+
+			await apiHelpers.headlessAdminUser.assignUserToSite(
+				siteRole.id,
+				site.id,
+				user.id
+			);
+			await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+				account.id,
+				[user.emailAddress]
+			);
 		});
 
-		await commerceAdminChannelsPage.goto();
-
-		await (
-			await commerceAdminChannelsPage.channelsTableRowLink(channel.name)
-		).click();
-
-		await commerceAdminChannelDetailsPage.maxOpenOrderAccountInput.fill(
-			'1'
-		);
-		await commerceAdminChannelDetailsPage.saveButton.click();
+		await test.step('Create an order via API and login as a Buyer', async () => {
+			order1 = await apiHelpers.headlessCommerceAdminOrder.postOrder({
+				accountId: account.id,
+				channelId: channel.id,
+				name: 'order1',
+				orderStatus: '2',
+			});
+
+			await page.waitForTimeout(2000);
+
+			await performLogout(page);
+			await performLoginViaApi({
+				page,
+				screenName: 'demo.unprivileged',
+			});
+		});
 
-		await page.goto(`/web/${site.name}`);
+		await test.step('Go to Site and check that the account selector has the order selected', async () => {
+			await page.goto(`/web/${site.name}`);
 
-		await commerceThemeMiniumCatalogPage.accountSelectorButton.click();
+			await page.waitForLoadState();
 
-		await expect(
-			await commerceThemeMiniumCatalogPage.createNewOrderButton
-		).toBeDisabled();
+			await expect(
+				commerceThemeMiniumCatalogPage.accountSelectorButton.locator(
```

Comment:
[NB] Check locators if there are double references, since here we removed the usage of Minium Site Init.